src/org/traccar/protocol/OwnTracksProtocolDecoder.java

```diff
@@ -58,6 +58,13 @@ protected Object decode(
         JsonObject root = Json.createReader(
                 new StringReader(request.getContent().toString(StandardCharsets.US_ASCII))).readObject();
 
+        if (!root.containsKey("_type") || !root.getString("_type").equals("location")) {
+            sendResponse(channel, HttpResponseStatus.BAD_REQUEST);
+            return null;
+        }
+
+        long timestamp;
+        String deviceId = new String();
```

Comment:
Why is there a `new String()` here. That's just wrong. Also, the name should be `uniqueId`. We use `deviceId` for internal device identifier. Also, why are these variables at the top?